MoltenVK/MoltenVK/GPUObjects/MVKDevice.mm

```diff
@@ -746,10 +750,25 @@
 	*(uint32_t*)&uuid[uuidComponentOffset] = NSSwapHostIntToBig(deviceID);
 	uuidComponentOffset += sizeof(deviceID);
 
-	// Last 8 bytes contain the GPU registry ID
-	uint64_t regID = mvkGetRegistryID(_mtlDevice);
-	*(uint64_t*)&uuid[uuidComponentOffset] = NSSwapHostLongLongToBig(regID);
-	uuidComponentOffset += sizeof(regID);
+	// Next 2 bytes contains GPU location
+	uint16_t dvcLoc = mvkGetLocation(_mtlDevice);
+	*(uint16_t*)&uuid[uuidComponentOffset] = NSSwapHostShortToBig(dvcLoc);
+	uuidComponentOffset += sizeof(deviceID);
+
+	// Next 2 bytes contains GPU location number
+	uint16_t dvcLocNum = mvkGetLocationNumber(_mtlDevice);
+	*(uint16_t*)&uuid[uuidComponentOffset] = NSSwapHostShortToBig(dvcLocNum);
+	uuidComponentOffset += sizeof(deviceID);
+
+	// Next 2 bytes contains GPU peer group ID
+	uint16_t pgID = mvkGetPeerGroupID(_mtlDevice);
+	*(uint16_t*)&uuid[uuidComponentOffset] = NSSwapHostShortToBig(pgID);
+	uuidComponentOffset += sizeof(deviceID);
```

Comment:
From what I've seen, all eight bytes of the peer group ID are significant.

Reply:
Fixed by hashing this as `uint64_t` into a final `uint64_t` ID that includes the other properties too.